pkg/plugin/backup_pvc_action_plugin.go

```diff
@@ -88,28 +90,45 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
 		}
 	}
 
+	objectToSnapshot := corev1.TypedLocalObjectReference{
+		APIGroup: &corev1.SchemeGroupVersion.Group,
+		Kind:     pvc.Kind,
+		Name:     pvc.Name,
+	}
+
 	backupRepositoryName, err := backupdriver.ClaimBackupRepository(ctx, utils.S3RepositoryDriver, repositoryParameters,
 		[]string{pvc.Namespace}, veleroNs, backupdriverClient, p.Log)
 	if err != nil {
 		p.Log.Errorf("Failed to claim backup repository: %v", err)
 		return nil, nil, errors.WithStack(err)
 	}
+	backupRepository := snapshotUtils.NewBackupRepository(backupRepositoryName)
 
-	objectToSnapshot := corev1.TypedLocalObjectReference{
-		APIGroup: &corev1.SchemeGroupVersion.Group,
-		Kind:     pvc.Kind,
-		Name:     pvc.Name,
+	p.Log.Info("Creating a Snapshot CR")
+	updatedSnapshot, err := snapshotUtils.SnapshotRef(ctx, backupdriverClient, objectToSnapshot, pvc.Namespace, *backupRepository,
+		[]backupdriverv1.SnapshotPhase{backupdriverv1.SnapshotPhaseSnapshotted, backupdriverv1.SnapshotPhaseSnapshotFailed}, p.Log)
+	if err != nil {
+		p.Log.Errorf("Failed to create a Snapshot CR: %v", err)
+		return nil, nil, errors.WithStack(err)
+	}
+	if updatedSnapshot.Status.Phase == backupdriverv1.SnapshotPhaseSnapshotFailed {
+		errMsg := fmt.Sprintf("Failed to create a Snapshot CR: Phase=SnapshotFailed, err=%v", updatedSnapshot.Status.Message)
+		p.Log.Error(errMsg)
+		return nil, nil, errors.New(errMsg)
 	}
 
-	p.Log.Info("Creating a snapshot CR")
-	backupRepository := snapshotUtils.NewBackupRepository(backupRepositoryName)
-	_, err = snapshotUtils.SnapshotRef(ctx, backupdriverClient, objectToSnapshot, pvc.Namespace, *backupRepository, []backupdriverv1api.SnapshotPhase{backupdriverv1api.SnapshotPhaseSnapshotted}, p.Log)
+	// Persist the snapshot blob as an annotation of PVC
+	snapshotAnnotation, err := pluginUtil.GetAnnotationFromSnapshot(updatedSnapshot)
 	if err != nil {
-		p.Log.Errorf("Failed to create a snapshot CR: %v", err)
+		p.Log.Errorf("Failed to marshal Snapshot object: %v", err)
 		return nil, nil, errors.WithStack(err)
 	}
+	vals := map[string]string{
+		utils.ItemSnapshotLabel: snapshotAnnotation,
```

Comment:
As discussed offline, the key here needs to be the snap-id.

Reply:
I just thought it twice and felt like using backupName might be better. The format of key would be like,
```
"velero-plugin-for-vsphere/item-snapshot-<Backup Name>"
```

Reply:
there could be multiple snaps right? a backup name may be unique.

Reply:
the pvc in object store is a per-backup object and the annotation will never be patched into PVC in cluster. So, we are good to stay the same and and use the constant annotation key.